mantidimaging/gui/windows/spectrum_viewer/view.py

```diff
@@ -267,21 +268,33 @@ def set_new_roi(self) -> None:
         """
         self.presenter.do_add_roi()
 
-    def update_roi_color_in_table(self, roi_name: str, new_color: tuple):
+    def handle_table_click(self, index):
+        if index.isValid() and index.column() == 1:
+            roi_name = self.roi_table_model.index(index.row(), 0).data()
+            self.spec_roi_color(roi_name)
+
+    def spec_roi_color(self, roi_name: str) -> None:
+        spectrum_roi = self.spectrum_widget.roi_dict.get(roi_name)
+        if spectrum_roi and hasattr(spectrum_roi, 'change_color_action'):
+            spectrum_roi.change_color_action.trigger()
+
+    def update_roi_color(self, roi_name: str, new_color: tuple) -> None:
         """
         Finds ROI by name in table and updates colour.
-
         @param roi_name: Name of the ROI to update.
         @param new_color: The new color for the ROI in (R, G, B) format.
         """
+        if roi_name in self.spectrum_widget.roi_dict:
+            self.spectrum_widget.roi_dict[roi_name].colour = new_color
+            self.spectrum_widget.roi_dict[roi_name].update()
```

Comment:
I don't think these lines are needed any more, as this gets done in `SpectrumROI.onChangeColor`